source/lac/petsc_precondition.cc

```diff
@@ -1094,43 +1096,106 @@ namespace PETScWrappers
     AssertThrow(ierr == 0, ExcPETScError(ierr));
   }
 
-  int
+#  ifndef PetscCall
+#    define PetscCall(code)             \
+      do                                \
+        {                               \
+          PetscErrorCode ierr = (code); \
+          CHKERRQ(ierr);                \
+        }                               \
+      while (0)
+#  endif
+
+  PetscErrorCode
+  PreconditionShell::pcsetup(PC ppc)
+  {
+    PetscFunctionBeginUser;
+    // Failed reason is not reset uniformly within the
+    // interface code of PCSetUp in PETSc.
+    // We handle it here.
+    PetscCall(PCSetFailedReason(ppc, PC_NOERROR));
```

Comment:
This breaks our regression testers, see https://cdash.dealii.org/viewBuildError.php?buildid=1876. Looks like this function was only added in PETSc 3.14, see https://github.com/petsc/petsc/commit/1b2b9847926d67f9d139cc21266b4ea2e2f01087 but we claim to support 3.7.

Reply:
@masterleinad https://github.com/dealii/dealii/pull/15368